uikit/viewmodels/src/main/java/com/sygic/maps/uikit/viewmodels/common/extensions/SdkExtensions.kt

```diff
@@ -173,4 +185,101 @@ fun Application.computePrimaryRoute(routePlan: RoutePlan, routeComputeCallback:
             override fun onPrimaryComputeFinished(router: Router, route: RouteInfo) = routeComputeCallback.invoke(route)
         })
     }
+}
+
+fun List<NaviSignInfo.SignElement>.concat(): String = StringBuilder().apply {
+    this@concat.forEach {
+        if (isNotEmpty()) append(", ")
+        append(it.text)
+    }
+}.toString()
+
+fun List<NaviSignInfo>.getNaviSignInfoOnRoute(): NaviSignInfo? {
+    this.filter { it.isOnRoute }.let { isOnRouteList ->
+        if (isOnRouteList.isEmpty()) {
+            return null
+        }
+
+        return isOnRouteList.firstOrNull { it.backgroundColor != 0 }?.let { it } ?: isOnRouteList[0]
+    }
+}
+
+fun NaviSignInfo.roadSigns(maxRoadSignsCount: Int = 3): List<RoadSignData> {
+    val roadSigns = mutableListOf<RoadSignData>()
+    val hasPic = signElements.find { it.elementType == SignElementType.Pictogram } != null
+    signElements
+        .asSequence()
+        .filter { it.elementType == SignElementType.RouteNumber }
+        .take(if (hasPic) maxRoadSignsCount - 1 else maxRoadSignsCount)
+        .toList()
+        .forEach { signElement ->
+            with(signElement.routeNumberFormat) {
+                if (insideNumber.isNotEmpty()) {
+                    roadSigns.add(
+                        RoadSignData(
+                            roadSignBackgroundDrawableRes(),
+                            insideNumber,
+                            roadSignForegroundColorRes()
+                        )
+                    )
+                }
+            }
+        }
+
+    return roadSigns
+}
+
+fun NaviSignInfo.createInstructionText(): TextHolder {
+    val acceptedSignElements = signElements
+        .filter { element -> element.elementType.let { it == SignElementType.ExitNumber || it == SignElementType.PlaceName || it == SignElementType.OtherDestination } }
+        .sortedByDescending { it.elementType == SignElementType.ExitNumber }
+
+    return if (acceptedSignElements.any { it.elementType == SignElementType.ExitNumber }) {
+        TextHolder.from(R.string.exit_number, acceptedSignElements.concat())
+    } else {
+        TextHolder.from(acceptedSignElements.concat())
+    }
+}
+
+fun DirectionInfo.createInstructionText(naviSignInfo: NaviSignInfo? = null): TextHolder {
```

Comment:
toto sa mi az tak nepozdava, ze byb extension funkcia `DirectionInfo` mala brat inu info `NaviSignInfo` a interne rozhodovat, co pouzije. Hlavne ked to niekedy nepotrebuje vobec a niekedy to pouzije iba `naviSignInfo`

Reply:
robi to vela veci v jednej funkcii, ktora by sa dala rozbit na 3

Reply:
oki, tak na to udělám utils metodu, když je problém s extension. to naviSignInfo a není to úplně unrelated třída a navíce je plně volitelná. volá se to ze dvou míst a nechci toto vnitřní rozhodování tahat ven. přál bych ti vidět tu logiku v SDK app, louskal jsem to dva dny než jsem to pochopil alespoň z 50%...
https://git.sygic.com/projects/NAVI/repos/android/browse/SygicNaviSdk/modules/signpostview/src/main/java/com/sygic/navi/navigation/viewmodel/signposts/DirectionsSignpostsViewModel.java

Reply:
no to ani necakm, ze je to tam krasne, ale aj tak sa mi nepaci take prepojenie 2 nezavislych data tried. ani v app asi nemali `direction.xyz(signpost)`.

A hlavne to co pises, ze je tam "rozhodovanie". Nejaka data trieda by nemala rozhodovat ako sa ma nieco vyskladat. V tej util to je lepsie, ale stale mi pride (aj ked na 2 miestach), ze kod 

```
{
    ...
    if direction.xyz then direction.xyz
    else if navi.xyz then navi.xyz
    else direction.abc
}

nie je az take opakovanie, ze by sme sa tomu za kazdu cenu mali vyhybat. a ked aj, tak tieto 3 riadky by mali byt v tej utils triede a ostatne uz mozu byt funkcie na tych klasach / extensiony